### DISCUSSION TITLE
Applying Recap decomposition on symmetric molecules

### DISCUSSION THREAD

**Discussion by Sowmya-R-Krishnan** [rdkit/rdkit#4819] [Q&A]
Hi all,

I tried applying Recap decomposition on a molecule dataset and came across a subset of molecules with two-fold symmetry. By two-fold symmetry, I mean the same sub-group/fragment occurring twice and connected by a small linker region (as in the example below). When applying Recap, the GetLeaves() function returns only one instance of the fragment for such molecules. Is there a way to know how many of these fragments are present in the molecule using Recap output or we can obtain this count only using an explicit substructure search for the fragment within the molecule?

Example molecule: C1CCc2nnc(-c3nnc4n3CCCCC4)n2CC1
Output from RDKit Recap module GetLeaves() function: ['*c1nnc2n1CCCCC2']

Noticed another example case: O=C(NN=Cc1ccc(O)c(S(=O)(=O)O)c1)NN=Cc1ccc(O)c(S(=O)(=O)O)c1
Here Recap GetLeaves() function returns: ['*NN=Cc1ccc(O)c(S(=O)(=O)O)c1'] - It is missing a ketone group present in the linker region connecting the fragments. Shouldn't the fragment with the ketone group be considered as a separate fragment?

Sowmya

**Comment by greglandrum** [comment]
@Sowmya-R-Krishnan,
I don't think doing a substructure match will help here since there could be other places in the molecule where the same fragments appear. Here's a simple example of that:
```
In [44]: m = Chem.MolFromSmiles('C1CC1Oc1cc(CCCOC2CC2)cc(OC2CC2)c1')

In [45]: d = Recap.RecapDecompose(m)

In [46]: d.GetLeaves()
Out[46]: 
{'*C1CC1': <rdkit.Chem.Recap.RecapHierarchyNode at 0x7f3afaf4f040>,
 '*CCCc1cc(*)cc(*)c1': <rdkit.Chem.Recap.RecapHierarchyNode at 0x7f3afaca4730>}
```
In this case the `C1CC1` fragment appears three times in the molecule, but only two of them are associated with the symmetric ring.

The RECAP results come back as a graph describing the decomposition and you can probably count the number of paths from the top of the tree to the bottom in order to determine how often a particular fragment comes from a particular group, but it's likely to be some work.

Maybe there's another way to do what you want: why are you interested in knowing this information?
